lib/ethpy/ethpy/hyperdrive/api/api_test.py

```diff
@@ -23,102 +25,208 @@
         # TODO: This should be exact up to 1e-18, but is not
-        fixed_rate = (FixedPoint(1) - spot_price) / (spot_price * hyperdrive.calc_position_duration_in_years())
-        assert abs(fixed_rate - hyperdrive.calc_fixed_rate()) <= FixedPoint(scaled_value=100)
+        fixed_rate = (FixedPoint(1) - spot_price) / (
+            spot_price * hyperdrive_interface.calc_position_duration_in_years()
+        )
+        assert abs(fixed_rate - hyperdrive_interface.calc_fixed_rate()) <= FixedPoint(scaled_value=100)
 
     def test_misc(self, local_hyperdrive_pool: DeployedHyperdrivePool):
-        """Placeholder for additional tests.
+        """Miscellaneous tests only verify that the attributes exist and functions can be called."""
+        hyperdrive_interface = self.setup_hyperdrive_interface(local_hyperdrive_pool)
+        _ = hyperdrive_interface.current_pool_state
+        _ = hyperdrive_interface.current_pool_state.variable_rate
+        _ = hyperdrive_interface.current_pool_state.vault_shares
+        _ = hyperdrive_interface.calc_open_long(FixedPoint(100))
+        _ = hyperdrive_interface.calc_open_short(FixedPoint(100))
+        _ = hyperdrive_interface.calc_bonds_given_shares_and_rate(FixedPoint(0.05))
+        _ = hyperdrive_interface.calc_max_long(FixedPoint(1000))
+        _ = hyperdrive_interface.calc_max_short(FixedPoint(1000))
+
+    def test_bonds_given_shares_and_rate(self, local_hyperdrive_pool: DeployedHyperdrivePool):
+        """Check that the bonds calculated actually hit the target rate."""
+        # pylint: disable=too-many-locals
+        hyperdrive_interface = self.setup_hyperdrive_interface(local_hyperdrive_pool)
+        # get pool config variables
+        pool_config = hyperdrive_interface.pool_config
+        init_share_price: FixedPoint = pool_config.initial_share_price
+        time_stretch: FixedPoint = pool_config.time_stretch
+        position_duration_years = pool_config.position_duration / FixedPoint(60 * 60 * 24 * 365)
+        # get pool info variables
+        pool_info = hyperdrive_interface.current_pool_state.pool_info
+        share_reserves: FixedPoint = pool_info.share_reserves
+        share_adjustment: FixedPoint = pool_info.share_adjustment
+        effective_share_reserves = share_reserves - share_adjustment
+        bond_reserves: FixedPoint = pool_info.bond_reserves
+        # check current rate
+        spot_price = ((init_share_price * effective_share_reserves) / bond_reserves) ** time_stretch
+        fixed_rate = (FixedPoint(1) - spot_price) / (spot_price * position_duration_years)
+        assert abs(fixed_rate - FixedPoint(0.05)) < FixedPoint(scaled_value=100)
+        # test hitting target of 10%
+        target_apr = FixedPoint("0.10")
+        bonds_needed = hyperdrive_interface.calc_bonds_given_shares_and_rate(target_rate=target_apr)
+        spot_price = ((init_share_price * effective_share_reserves) / bonds_needed) ** time_stretch
```

Comment:
I think it would be better to modify the pool state directly and then pass it to calc_fixed_rate

Reply:
removed a bunch of stuff and passing pool_state directly to calc_fixed_rate as discussed, in 25690e3a240271b20a7765d72deb0cfc4b19effd